mrjones2014/legendary.nvim
# Custom sorting

Resolves: #218 

## How to Test

Excerpt from `config.lua` updates:

```lua
  -- Options for list sorting. Note that fuzzy-finders will still
  -- do their own sorting. For telescope.nvim, you can set it to use
  -- `require('telescope.sorters').fuzzy_with_index_bias({})` when
  -- triggered via `legendary.nvim`. Example config for `dressing.nvim`:
  --
  -- require('dressing').setup({
  --  select = {
  --    get_config = function(opts)
  --      if opts.kind == 'legendary.nvim' then
  --        return {
  --          telescope = {
  --            sorter = require('telescope.sorters').fuzzy_with_index_bias({})
  --          }
  --        }
  --      else
  --        return {}
  --      end
  --    end
  --  }
  -- })
  sort = {
    -- sort most recently used item to the top
    most_recent_first = true,
    -- sort user-defined items before built-in items
    user_items_first = true,
    -- sort the specified item type before other item types,
    -- value must be one of: 'keymap', 'command', 'autocmd', nil
    item_type_bias = nil,
  }
```

So, set up legendary like:

```lua
require('legendary').setup({
  sort = {
    most_recent_first = true,
    user_items_first = true,
    item_type_bias = 'command',
  }
})
```

Then set up `dressing.nvim` as shown in the comment above.

Then, items should be sorted by:

- most recent item first
- user-defined items before built-ins
- commands before other item types

## Testing for Regressions

I have tested the following:
